### DISCUSSION TITLE
Be able to set Tickrate of a BeehaveTree

### DISCUSSION THREAD

**Discussion by Werninator** [bitbrain/beehave#174] [Ideas]
My game will have hundreds of active BeehaveTrees running at once. Some trees don't need to run multiple times per second (maybe an Animal AI that just walks around its perimiter). Or a villager that is offscreen and doesn't need to do the most efficient work.

**Describe the solution you'd like**
Be able to set the Tickrate of a BeehaveTree - maybe even dynamically.

**Additional context**
This could also help with slowing down the tree to debug things like stated in #141 point 3.

**Comment by bitbrain** [comment]
The same can be technically achieved by changing the tick rate of your game itself. How exactly would this functionality differ?

**Reply by Werninator** [reply]
Uhh.. well. I want to be able to set variable tick rates depending on where the camera is currently located. In my case I will have a map with a fairly decent size.

**Reply by bitbrain** [reply]
Good point. I found this proposal on it: https://github.com/godotengine/godot-proposals/issues/439

**Reply by TrayserCassa** [reply]
Hey, I also stumbled across this proposal, but it's already 3y old. My idea would be to add an export for a time. If the time is set, we will use a timer. If the time value is empty the process function is used.

I currently use this for attacking the player and control the speed of each beehavetree let me easily define an attack rate.
